### PR TITLE
fix (fxa-settings) : render localized {$retryAfter} variable

### PR DESCRIPTION
**Pull request by vpomerleau**
## Because

* We want to show a localized time instead of {$retryAfter} variable in "You've tried too many times. Please try again {$retryAfter}" error message for the ResetPassword page.

## This pull request

* Adjust localized error message to include localized time.
* Fix test to reflect the actual auth errors that are now rendered instead of a generic error.

Edited to add:
* Adjust l10n comment for error string - some misunderstandings were identified in existing translations.
* Add missing FTL string for input label.

## Issue that this pull request solves

Closes: #FXA-6875

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

### CODE REVIEW

File: packages/fxa-settings/src/pages/ResetPassword/en.ftl
Comment (vpomerleau) [inline packages/fxa-settings/src/pages/ResetPassword/en.ftl:14]: @millsoper I'm noticing now that this success message used to be displayed in an alertBar when navigating to the next page, but has been removed in a previous commit and is no longer used anywhere. I'm not sure if this is something we might want to reintroduce or is ok to drop.

**Comment by bcolsson** [comment]
Can you confirm this change doesn't make any changes to how the strings are rendered currently?
(Basically can you confirm that the current implementation for localized strings returns `retryAfterLocalized` not `retryAfter`?)

That being said, looking at how languages localized this it seems a good portion of them didn't quite understand how the text would appear (so I see a lot that could potentially show up as "in in 15 minutes"). Would you be able to provide screenshots of how it appears in languages like Japanese and Italian?
If so, we might take this opportunity to update the string with better commenting to fix this issue.

**Comment by vpomerleau** [comment]
Really good questions @bcolsson!

> Can you confirm this change doesn't make any changes to how the strings are rendered currently? (Basically can you confirm that the current implementation for localized strings returns `retryAfterLocalized` not `retryAfter`?)

The problem we're currently seeing on stage is this: 

image

Unfortunately I haven't been able to reproduce this throttling error on local, and I haven't succeeded in mocking this variable localization in storybook (even for other pages where this error message is correctly localized on stage). The solution in this PR is based on similar implementation from Settings' `PageSecondaryEmailAdd` (see the rendered error messages below on stage for that page) and `PageRecoveryKeyAdd` - it should work, but I'll check back once this hits stage (this page won't be going into production for at least another sprint, so we have some time to adjust).

> That being said, looking at how languages localized this it seems a good portion of them didn't quite understand how the text would appear (so I see a lot that could potentially show up as "in in 15 minutes"). Would you be able to provide screenshots of how it appears in languages like Japanese and Italian? If so, we might take this opportunity to update the string with better commenting to fix this issue.

Here's an example of the same localized error message as rendered in French, Japanese and Italian on the `PageSecondaryEmailAdd`. I see what you mean about the doubled "tra" in italian, though French localizes correctly and... I don't know about Japanese). It might be helpful to reference the library that we use to localize these timestamps:
https://momentjs.com/docs/#/displaying/fromnow/

I've updated the PR with an adjustment to the l10n comment for this error string - please have a look and let me know if this seems clearer for our localizers. If so, would you be able to notify teams of this change once this is approved and merged?

image
 
image

image

**Comment by flodolo** [comment]
The one in Italian was definitely wrong :-(

**Comment by bcolsson** [comment]
Thanks for confirming how this renders @vpomerleau . Japanese also is incorrect, the equivalent to `in` repeats twice (`後後`).
I did spend some time investigating how moments.js works yesterday and this seems to confirm my understanding how it works. While it doesn't state it in the documentation if the time is in the future fromNow() pulls the `future` case from the appropriate locales [here](https://github.com/moment/moment/tree/develop/locale).

Unfortunately, as we can see in the state of the string, this isn't very straight forward for localizers to understand how this will get rendered. The other unfortunate side-effect is that if moments.js doesn't support a locale it will result in the string displaying a mix of languages. (Example: Guarani (gn))

Screenshot 2023-03-14 at 09 55 58

Considering the prominence of the string, I won't hold up this PR but we should consider alternatives such as just passing the number of minutes and using Fluent variants to display the message like:
```
auth-error-114 =
    { $retryAfter ->
        [one] You’ve tried too many times. Please try again in { $retryAfter } minute.
       *[other] You’ve tried too many times. Please try again in { $retryAfter } minutes.
    }
```

**Review by bcolsson — APPROVED** [review]

**Comment by vpomerleau** [comment]
> Unfortunately, as we can see in the state of the string, this isn't very straight forward for localizers to understand how this will get rendered. The other unfortunate side-effect is that if moments.js doesn't support a locale it will result in the string displaying a mix of languages. (Example: Guarani (gn))
>[...] we should consider alternatives such as just passing the number of minutes and using Fluent variants to display the message like:
> 
> ```
> auth-error-114 =
>     { $retryAfter ->
>         [one] You’ve tried too many times. Please try again in { $retryAfter } minute.
>        *[other] You’ve tried too many times. Please try again in { $retryAfter } minutes.
>     }
> ```

I agree this is not ideal, but it would not be a simple fix because this variable is used in various scenarios where the time frame is not always in minutes. I'll file a ticket with your observations so that the Eng team can discuss (I don't have complete knowledge of the cases where this variable is used).

**Comment by vpomerleau** [comment]
@bcolsson Issue filed: https://mozilla-hub.atlassian.net/browse/FXA-7046

**Comment by bcolsson** [comment]
@vpomerleau - thanks for filing the issue. I've fixed or asked localizers to review translations for potentially affected locales.

**Pull request merged**
